CONTRIBUTING.md

```diff
@@ -0,0 +1,205 @@
+# Contributing to HClib
+
+Before opening a pull request,
+please be sure to read through these guidelines.
+Following the guidelines listed below will make it much more likely
+for your contributions to be quickly accepted and merged.
+
+## Licensing
+
+All contributions submitted to this project are governed
+by the terms outlined in our license agreement.
+Specifically, our license (Apache 2.0)
+has an explicit contributions clause in [section
+5](https://github.com/habanero-rice/hclib/blob/6ad778a/LICENSE#L131-L137):
+
+> Unless You explicitly state otherwise, any Contribution intentionally
+> submitted for inclusion in the Work by You to the Licensor shall be under
+> the terms and conditions of this License, without any additional terms or
+> conditions.  Notwithstanding the above, nothing herein shall supersede or
+> modify the terms of any separate license agreement you may have executed
+> with Licensor regarding such Contributions.
+
+## Pull Requests
+
+Please follow established best practices for your pull requests.
+Give a short description of *what* your patch does in the title,
```

Comment:
e.g. below it says "A complex title or summary is a good indication that your pull request is doing too many things, and should be split into multiple patches" when it seems like it would be simpler to just say "A complex title or summary is a good indication that your pull request is doing too many things, and should be split into multiple pull requests"